pyAudioAnalysis/audioTrainTest.py

```diff
@@ -501,20 +501,35 @@ def feature_extraction_train_regression(folder_name, mid_window, mid_step,
     return errors, errors_base, best_params
 
 
-def load_model_knn(knn_model_name, is_regression=False):
-    with open(knn_model_name, "rb") as fo:
-        features = cPickle.load(fo)
-        labels = cPickle.load(fo)
-        mean = cPickle.load(fo)
-        std = cPickle.load(fo)
+def load_model_knn(knn_model_name, is_regression=False, mf_object=None):
+    if mf_object is not None:
+        mf_object.seek(0)
+        features = cPickle.load(mf_object)
+        labels = cPickle.load(mf_object)
+        mean = cPickle.load(mf_object)
+        std = cPickle.load(mf_object)
         if not is_regression:
-            classes = cPickle.load(fo)
-        neighbors = cPickle.load(fo)
-        mid_window = cPickle.load(fo)
-        mid_step = cPickle.load(fo)
-        short_window = cPickle.load(fo)
-        short_step = cPickle.load(fo)
-        compute_beat = cPickle.load(fo)
+            classes = cPickle.load(mf_object)
+        neighbors = cPickle.load(mf_object)
+        mid_window = cPickle.load(mf_object)
+        mid_step = cPickle.load(mf_object)
+        short_window = cPickle.load(mf_object)
+        short_step = cPickle.load(mf_object)
+        compute_beat = cPickle.load(mf_object)
+    else:
+        with open(knn_model_name, "rb") as fo:
+            features = cPickle.load(fo)
+            labels = cPickle.load(fo)
+            mean = cPickle.load(fo)
+            std = cPickle.load(fo)
+            if not is_regression:
+                classes = cPickle.load(fo)
+            neighbors = cPickle.load(fo)
+            mid_window = cPickle.load(fo)
+            mid_step = cPickle.load(fo)
+            short_window = cPickle.load(fo)
+            short_step = cPickle.load(fo)
+            compute_beat = cPickle.load(fo)
```

Comment:
This is the place I meant talking about large c&p.

Maybe, not to pollute global namespace, Python can have inner def-s?

```
def outer(...):
  def inner(...):
    ...
  foo = inner(...)
```

?